lib/effects/effects.dart

```diff
@@ -157,22 +157,31 @@ abstract class PositionComponentEffect
     endSize = component.size.clone();
   }
 
-  @override
-  void setComponentToOriginalState() {
+  /// Only change the parts of the component that is affected by the
+  /// effect, and only set the state if it is the root effect (not part of
+  /// another effect, like children of a CombinedEffect or SequenceEffect).
+  void _setComponentState(Vector2 position, double angle, Vector2 size) {
     if (isRootEffect()) {
-      component?.position?.setFrom(originalPosition);
-      component?.angle = originalAngle;
-      component?.size?.setFrom(endSize);
+      if (originalPosition != endPosition) {
```

Comment:
This if shouldn't include the `position` variable?

Reply:
No, it is just checking whether this effect made any changes on the component.
The position variable will be either `endPosition` or `originalPosition`, but this way I don't have to duplicate the code.